### PR TITLE
Add dict chemicals

### PR DESCRIPTION
**Pull request by marikaris**

### CODE REVIEW

File: R/reshape.R [hunk 1]
```diff
@@ -31,15 +32,19 @@ du.reshape <- function(upload = TRUE, project, data_version, input_format, dict_
...
+   data, dict_kind
   )
   if (!is.null(yearlyrep_data)) write_csv(yearlyrep_data, paste0(getwd(), "/", file_name_yearly, ".csv"), na = "")
-  monthlyrep_data <- du.reshape.generate.monthly.repeated(
-    data, dict_kind
-  )
-  if (!is.null(monthlyrep_data)) write_csv(monthlyrep_data, paste0(getwd(), "/", file_name_monthly, ".csv"), na = "")
+  #monthlyrep_data <- du.reshape.generate.monthly.repeated(
+  # data, dict_kind
+  #)
```
Comment (marikaris) [inline R/reshape.R:43]: Do we remove the reshape for the monthly rep altogether? is this what we want?

File: R/reshape.R [hunk 2]
```diff
@@ -89,7 +94,7 @@ du.reshape <- function(upload = TRUE, project, data_version, input_format, dict_
     if (ds_upload.globals$login_data$driver == du.enum.backends()$OPAL) {
       if (!is.null(nonrep_data)) du.opal.upload(dict_kind, file_name_nonrep)
```
Comment (marikaris) [inline R/reshape.R:95]: these `is.null`'s are correct?

File: R/reshape_helpers.R [hunk 3]
```diff
@@ -22,7 +22,9 @@ du.read.source.file <- function(input_path, input_format) {
     } else if (input_format == du.enum.input.format()$R) {
       data <- source(input_path)
     } else {
-      data <- read_csv(input_path, col_types = cols(.default = col_double()))
+
+      #data <- read_csv(input_path, col_types = cols(.default = col_double()))
```
Comment (marikaris) [inline R/reshape_helpers.R:26]: remove uncommented line of code

File: R/upload.R [hunk 4]
```diff
@@ -69,11 +69,18 @@ du.upload <- function(dict_version = "2_1", data_version = "1_0", dict_kind = du
...
       paste(ds_upload.globals$dictionaries_outcome, collapse = ", ")
     )
+  } else if (dict_version != "" && dict_kind == du.enum.dict.kind()$CHEMICALS && !(dict_version %in% ds_upload.globals$dictionaries_chemicals)) {
+    stop(
+      "Version: [ ", dict_version, " ] is not available in published data dictionaries. Possible dictionaries are: ",
+      paste(ds_upload.globals$dictionaries_chemicals, collapse = ", ")
+    )
   } else {
     if (dict_version == "" && dict_kind == du.enum.dict.kind()$CORE) {
       dict_version <- "2_1"
```
Comment (marikaris) [inline R/upload.R:79]: This means the core variables of >2.1 don't work either, right? Interesting, we're supposed to be at 2.3 for core and 1.6 for outcome.

File: R/dictionaries.R [hunk 5]
```diff
@@ -106,8 +106,10 @@ du.populate.dict.versions <- function(dict_kind, dict_version) {
   
   if (dict_kind == du.enum.dict.kind()$CORE) {
     ds_upload.globals$dictionaries_core <- versions
-  } else {
+  } else if (dict_kind == du.enum.dict.kind()$OUTCOME) {
     ds_upload.globals$dictionaries_outcome <- versions
+  } else if (dict_kind == du.enum.dict.kind()$CHEMICALS) {
+    ds_upload.globals$dictionaries_chemicals <- versions
   }
```
[the remark this answers is not in the document]
Comment (erikzwart) [reply R/dictionaries.R:115]: added stop if dictionary is not supported

File: R/reshape.R [hunk 6]
```diff
@@ -31,15 +32,19 @@ du.reshape <- function(upload = TRUE, project, data_version, input_format, dict_
   nonrep_data <- du.reshape.generate.non.repeated(
     data, dict_kind
   )
+  #if (!is.null(nonrep_data)) {write_csv(nonrep_data, paste0(getwd(), "/", file_name_nonrep, ".csv"), na = "")}
```
[the remark this answers is not in the document]
Comment (erikzwart) [reply R/reshape.R]: removed line

File: R/reshape.R [hunk 7]
```diff
@@ -31,15 +32,19 @@ du.reshape <- function(upload = TRUE, project, data_version, input_format, dict_
   nonrep_data <- du.reshape.generate.non.repeated(
     data, dict_kind
   )
+  #if (!is.null(nonrep_data)) {write_csv(nonrep_data, paste0(getwd(), "/", file_name_nonrep, ".csv"), na = "")} 
   if (!is.null(nonrep_data)) write_csv(nonrep_data, paste0(getwd(), "/", file_name_nonrep, ".csv"), na = "")
```
[the remark this answers is not in the document]
Comment (erikzwart) [reply R/reshape.R]: replaced is.null by exists and added {} brackets

File: R/reshape.R (hunk 1 above)
Comment (erikzwart) [reply R/reshape.R]: uncommented the code and replaced is.null by exists

File: R/reshape.R (hunk 2 above)
Comment (erikzwart) [reply R/reshape.R]: replaced is.null by exists

File: R/reshape_helpers.R (hunk 3 above)
Comment (erikzwart) [reply R/reshape_helpers.R]: removed line

File: R/upload.R (hunk 4 above)
Comment (erikzwart) [reply R/upload.R:79]: If dict_version is empty the default version will be set to 2_1, you can set this to 1_0 or 2_0

**Pull request merged**